pkg/registry/apps/statefulset/strategy.go

```diff
@@ -127,6 +127,10 @@ func dropStatefulSetDisabledFields(newSS *apps.StatefulSet, oldSS *apps.Stateful
 			newSS.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable = nil
 		}
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetSlice) {
+		// Reset Spec.Ordinals to the default value (nil).
+		newSS.Spec.Ordinals = nil
```

Comment:
You also need to set the strategy for Update. And add tests.

Reply:
This function "dropStatefulSetDisabledFields" is called by "PrepareForCreate" and "PrepareForUpdate", so I think this covers both strategies for update and create. Let me know if I'm missing something.

Added tests.

Reply:
That is odd. It sounds inconsistent with how we do updates anywhere else.

See step 3 here:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-field-in-existing-api-version

Reply:
Ah, I see, updated the logic and tests to ensure existing data is preserved when the feature gate is disabled. One thing I wasn't clear about was how to handle updates if the field is already in use. Based on the example, if `.newSS.spec.Ordinals` is set, the persisted object will get updated even when the feature gate is disabled. However, I'm not sure if this is correct, should we just retain the `.oldSS.spec.Ordinals`?

Reply:
if the feature is disabled, but the old object already used the field, it should be retained or even modified.

This is necessary when, during a control plane upgrade, an apiserver is one version ahead where the feature is enabled. This apiserver could have accepted the field. Old apiservers should respect what the newer apiserver did.